promql/lex.go

```diff
@@ -889,3 +889,16 @@ func isDigit(r rune) bool {
 func isAlpha(r rune) bool {
 	return r == '_' || ('a' <= r && r <= 'z') || ('A' <= r && r <= 'Z')
 }
+
+// isLabel reports whether the string can be used as label.
+func isLabel(s string) bool {
```

Comment:
Wouldn't it be more canonical to just match against `model.LabelNameRE`? https://github.com/prometheus/common/blob/master/model/labels.go#L84